src/Microsoft.VisualStudio.Composition/Configuration/AttributedPartDiscovery.cs

```diff
@@ -297,8 +303,17 @@ private bool TryCreateImportDefinition(Type importingType, ICustomAttributeProvi
                 sharingBoundaries = sharingBoundaries.Union(sharingBoundaryAttribute.SharingBoundaryNames);
             }
 
+            if (member is PropertyInfo importingMember && importingMember.SetMethod == null)
+            {
+                // MEFv2 quietly ignores such importing members.
+                importDefinition = null;
+                return false;
+            }
+
             if (importAttribute != null)
             {
+                this.ThrowOnInvalidImportingMemberOrParameter(member);
```

Comment:
As currently written, you're right. However this method may add more validation checks in the future that would lead to the part being rejected.
The general idea I'm using here is that VS MEF knows that at runtime, it requires certain patterns to be true (e.g. importing properties have setters) and if those conditions don't hold, the part should be rejected. As this PR shows, we're only discovering some of these requirements now, so this method is a placeholder for all such conditions we learn in the future. And for this particular derived class, we know that this particular condition should be responded to different for MEFv2 attributes, so we skip this 'invalid' aspect in order to emulate MEFv2 behavior for this particular case.